### ISSUE TITLE
Rule proposal: 

### ISSUE THREAD

**Issue by 4dvn-tam** [typescript-eslint/typescript-eslint#4244]
### Description

typescript-eslint/strict-boolean-expressions

When a variable has multiple types.
let something : string | boolean | undefined;

There is no config to allow this type in condition

I tried:
allowString: true,
allowNullableString: true,
allowNullableBoolean: true,

but they did not work together

### Fail

```ts
let something: string | boolean | undefined;
if(something) { ... }
```

### Pass

There should be an option to allow
```ts
let something: string | boolean | undefined;
if(something) { ... }
```

**Comment by bradzacher** [comment]
Ignoring the `undefined` - for `string | boolean` you have: 
- two truthy states: "non-empty string" and `true`
- two falsey states: "empty string" and `false`

If you do `if (something)` what does that mean exactly? Does that mean you're intentionally treating both truthy states the same? And you're treating both falsey states the same?
How does someone new that comes to your code know that is your intention?

The rule is about being strict with your boolean conditions.
When there is more than one state, the rule enforces that you handle it explicitly.
This is one of the invariants in the rule for which we won't be adding an option.
